File: programmers/python 3/12943.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12943

def solution(num):
    for count in range(500):
        if num == 1:
            return count
        num = num * 3 + 1 if num % 2 else num // 2
    return -1

File: programmers/swift/12943.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12943

func solution(_ num:Int) -> Int {
    var num = num
    for count in 0 ..< 500 {
        if num == 1 {
            return count
        }

        num = num % 2 == 0 ? num / 2 : num * 3 + 1
    }
    return -1
}